---
title: "Hierarchical nonparametric regression"
author: "Spencer Woody"
date: 2018-01-30
categories: ["Project", "R", "wip"]
tags: []
summary: Hierarchical Gaussian process regression with application to drosophila microarray data
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE)
```

## Introduction

Hierarchical Gaussian process regression with application to drosophila microarray data, implemented in R and visualized using `ggplot2`. 

Based on a paper by [Hensman, Lawrence, and Rattray](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-252).

## Read in the data

```{r data, message=F, warning=F}
library(dplyr)
library(forcats)
library(ggplot2)
theme_set(theme_bw(base_size = 12))

drosophila <- read.csv("data/drosophila.csv")

drosophila <- drosophila %>%
  mutate(gene = fct_inorder(gene))
```

# Plot the data

```{r, make_plot}
allgenes <- drosophila %>% 
    ggplot(aes(time, log2exp)) +
    geom_point(aes(col = group, pch = replicate)) +
    facet_wrap(~gene) +
    labs(title = "Expression Profiles for all Genes",
         x = "Time",
         y = "log2-Normalized gene expression") + 
    scale_colour_brewer(palette = "Dark2") +
    scale_x_continuous(breaks = seq(0, 12, by = 2)) 
```


You can also embed plots. See Figure \@ref(fig:facet1) for example:

```{r facet1, fig.cap='faceted plot of all genes', tidy=FALSE, echo = F}
allgenes
```
